src/agent/scm/lib/svnwrapper.ts

```diff
@@ -0,0 +1,475 @@
+        // default connection related args
+        var usernameArg = '--username';
+        var passwordArg = '--password';
+        
+        var defaults: string[] = [];
+        
+        if (this.endpoint.username && this.endpoint.username.length > 0) {
+            this.ctx.verbose("username=" + this.endpoint.username);
+            defaults.push(usernameArg, this.endpoint.username);
+        }
+        if (this.endpoint.password && this.endpoint.password.length > 0) {
+            this.ctx.verbose("password=" + this.endpoint.password);
+            defaults.push(passwordArg, this.endpoint.password);
+        }
+
+        var quotedArg = function(arg) {
+            var quote = '"';
+            if (arg.indexOf('"') > -1) {
+                quote = '\'';
+            }
+            return quote + arg + quote;
+        }
+
+        return args.concat(defaults).map((a) => quotedArg(a));
+    }
+
+    private _scrubCredential(msg: string): string {
+        if (msg && typeof msg.replace === 'function' 
+                    && this.endpoint.password) {
+            return msg.replace(this.endpoint.password, "******************");
+        }
+
+        return msg;
+    }
+
+    private _exec(cmd: string, args: string[], options?: ISvnExecOptions): Q.Promise<number> {
+        if (this.svnPath === null) {
+            return this._getSvnNotInstalled();
+        }
+
+        var svn = new tl.ToolRunner(this.svnPath);
+        svn.silent = !this.isDebugMode();
+
+        svn.on('debug', (message) => {
+            this.emit('stdout', '[debug]' + this._scrubCredential(message));
+        })
+
+        svn.on('stdout', (data) => {
+            this.emit('stdout', this._scrubCredential(data));
+        })
+
+        svn.on('stderr', (data) => {
+            this.emit('stderr', this._scrubCredential(data));
+        })
+
+        // cmd
+        svn.arg(cmd, true);
+
+        var quotedArgs = this._getQuotedArgsWithDefaults(args);
+        // args
+        if (quotedArgs.map((arg: string) => {
+            svn.arg(arg, true); // raw arg
+        }));
+
+        options = options || <ISvnExecOptions>{};
+        var ops: any = {
+            cwd: options.cwd || process.cwd(),
+            env: options.env || process.env,
+            silent: !this.isDebugMode(),
+            outStream: options.outStream || process.stdout,
+            errStream: options.errStream || process.stderr,
+            failOnStdErr: options.failOnStdErr || false,
+            ignoreReturnCode: options.ignoreReturnCode || false
+        };
+
+        return svn.exec(ops);
+    }
+
+    private _shellExec(cmd, args: string[]): Q.Promise<any> {
+        if (this.svnPath === null) {
+            return this._getSvnNotInstalled();
+        }
+
+        var cmdline = this.svnPath + ' ' + cmd + ' ' + this._getQuotedArgsWithDefaults(args).join(' ');
+        
+        return utilm.exec(cmdline)
+        .then ((v) => {
+            this.ctx.verbose(cmdline);
```

Comment:
Currently nothing does, but I'll add scrubbing for any case, to be safe if in the future the function usage changes.